include/aie/Dialect/AIE/IR/AIEInterfaces.td

```diff
@@ -134,6 +134,15 @@ def AIETarget : OpInterface<"AIETarget"> {
 }
 
 // Don't delete - see AIEDialect::myVerifyOffsetSizeAndStrideOp
-def MyOffsetSizeAndStrideOpInterface: OpInterfaceTrait<"::xilinx::AIE::MyOffsetSizeAndStrideOpInterface"> {}
+def MyOffsetSizeAndStrideOpInterface : OpInterfaceTrait<"::xilinx::AIE::MyOffsetSizeAndStrideOpInterface"> {}
+
+// Ops may specify that the accessibility checks of their child ops are not verified.
+// For example, by default, ops will verify that all contained aie.buffers are
+// accessible from within the tile from which they are used. However, in abstract
+// definitions not tied to any tile, it may still be useful to be able to refer
+// to buffers and skip those checks.
+def SkipAccessibilityCheckTrait : NativeOpTrait<"SkipAccessibilityCheckTrait"> {
+  string cppNamespace = "::xilinx::AIE";
+}
```

Comment:
If the op is associated with a buffer, then isn't it already associated with at tile? Can you give an example of where this is not true? It seems like there should be a simpler way to to verification.

Reply:
> If a op refers a buffer, then isn't it already associated with at tile?
This is true for all ops except the newly introduced `aie.bd_chain`.

> Can you give an example of where this is not true? 
`aie.bd_chain` chains are not associated with any tile _until_ they are concretized (started) on a particular tile, channel, direction.

Before this change, it would be impossible to define an abstract BD chain that referenced a buffer on a tile. After this, you can describe the abstract chain.

The compiler will then throw an error if you try to start that BD chain on a tile that does not have access to the buffer defined within the chain, or allow it if you start it on a 

Here's an example:

```
%some_tile = aie.tile(1, 2)
%other_tile = aie.tile(3, 2)
%buffer_on_some_tile = aie.buffer(%some_tile)

// Abstract BD chain _definition_ should be allowed:
aie.bd_chain @abstract(...) {
  // ...
  aie.dma_bd(%buffer_on_some_tile)
  // ...
}

// Once we concretize it ...
aiex.runtime_sequence () {
  // ... this should be allowed:
  aiex.start_bd_chain @abstract(...) on (%some_tile, MM2S, 0)  
   // ... but this will throw an error, because a buffer referenced within the chain is not accessible on %other_tile:
   aiex.start_bd_chain @abstract(...) on (%other_tile, MM2S, 0)
}
```

Reply:
Got it. The purpose of `SkipAccessibilityCheckTrait` is to exempt `bd_chain` from the requirement that buffers and locks are contained in a `TileElement`. `AIE::DeviceOp` and `ModuleOp` are also exempt if they are the immediate parent but we don't (can't) add the trait to those. All because `bd_chain` can't be a `TileElement` since it can exist outside a `TileOp`. `TileElement`'s only use is to provide tile id and asm printing. If `getTileID` could return `std::optional<AIE::TileID>` then could `bd_chain` be a (logical) `TileElement`?

Reply:
Yes, that would be an alternative way to go about this. Then we would need to update the verifier to instead of checking for that trait, checks if `getTileID` returned an empty optional.

Any reason you would prefer this?

It would feel weird to me to make `bd_chain` a `TileElement`, since it can be instantiated on other tiles as well.

Reply:
> Any reason you would prefer this?

In general I would prefer to enable logical cores, buffers, locks, etc, but obviously we won't do it in this PR.